Zeronsd does not compile due to update with clap

Since clap updated to a new major verison yesterday, Zeronsd does not compile correctly. 
The specific errors are 
```
error[E0107]: this struct takes 0 lifetime arguments but 1 lifetime argument was supplied
  --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/zeronsd-0.2.4/src/supervise.rs:91:18
   |
91 | impl From<&clap::ArgMatches<'_>> for Properties {
   |                  ^^^^^^^^^^---- help: remove these generics
   |                  |
   |                  expected 0 lifetime arguments
   |
note: struct defined here, with 0 lifetime parameters
  --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.0.0/src/parse/matches/arg_matches.rs:71:12
   |
71 | pub struct ArgMatches {
   |            ^^^^^^^^^^

error[E0107]: this struct takes 0 lifetime arguments but 1 lifetime argument was supplied
  --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/zeronsd-0.2.4/src/main.rs:39:29
   |
39 | fn unsupervise(args: &clap::ArgMatches<'_>) -> Result<(), anyhow::Error> {
   |                             ^^^^^^^^^^---- help: remove these generics
   |                             |
   |                             expected 0 lifetime arguments
   |
note: struct defined here, with 0 lifetime parameters
  --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.0.0/src/parse/matches/arg_matches.rs:71:12
   |
71 | pub struct ArgMatches {
   |            ^^^^^^^^^^

error[E0107]: this struct takes 0 lifetime arguments but 1 lifetime argument was supplied
  --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/zeronsd-0.2.4/src/main.rs:43:27
   |
43 | fn supervise(args: &clap::ArgMatches<'_>) -> Result<(), anyhow::Error> {
   |                           ^^^^^^^^^^---- help: remove these generics
   |                           |
   |                           expected 0 lifetime arguments
   |
note: struct defined here, with 0 lifetime parameters
  --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.0.0/src/parse/matches/arg_matches.rs:71:12
   |
71 | pub struct ArgMatches {
   |            ^^^^^^^^^^

error[E0107]: this struct takes 0 lifetime arguments but 1 lifetime argument was supplied
  --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/zeronsd-0.2.4/src/main.rs:47:23
   |
47 | fn start(args: &clap::ArgMatches<'_>) -> Result<(), anyhow::Error> {
   |                       ^^^^^^^^^^---- help: remove these generics
   |                       |
   |                       expected 0 lifetime arguments
   |
note: struct defined here, with 0 lifetime parameters
  --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.0.0/src/parse/matches/arg_matches.rs:71:12
   |
71 | pub struct ArgMatches {
   |            ^^^^^^^^^^

error[E0107]: this struct takes 0 lifetime arguments but 1 lifetime argument was supplied
  --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/zeronsd-0.2.4/src/supervise.rs:92:26
   |
92 |     fn from(args: &clap::ArgMatches<'_>) -> Self {
   |                          ^^^^^^^^^^---- help: remove these generics
   |                          |
   |                          expected 0 lifetime arguments
   |
note: struct defined here, with 0 lifetime parameters
  --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.0.0/src/parse/matches/arg_matches.rs:71:12
   |
71 | pub struct ArgMatches {
   |            ^^^^^^^^^^

For more information about this error, try `rustc --explain E0107`.
```

for now I installed with `cargo install --locked zeronsd`

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,7 +17,7 @@ edition = "2018"
 [dependencies]
 regex = ">= 0"
 anyhow = ">= 0"
-clap = ">= 0"
+clap = "^2"
 ipnetwork = ">= 0"
 trust-dns-resolver = "0.20.2"
 trust-dns-server = { version = "0.20.2", features = ["trust-dns-resolver"] } 

